dolphinscheduler-tools/src/main/java/org/apache/dolphinscheduler/tools/demo/ProcessDefinitionDemo.java

```diff
@@ -0,0 +1,778 @@
+        taskDefinitionLogSecond.setTaskParams("{\"localParams\":[],\"rawScript\":\"echo \\\"default\\\"\",\"resourceList\":[]}");
+        taskDefinitionLogSecond.setTaskType("SHELL");
+
+        TaskDefinitionLog taskDefinitionLogThird = taskDefinitionLogs.get(2);
+        taskDefinitionLogThird.setCode(taskCodes.get(2));
+        taskDefinitionLogThird.setName("TaskA");
+        taskDefinitionLogThird.setDescription("execute TaskA");
+        taskDefinitionLogThird.setTaskParams("{\"localParams\":[],\"rawScript\":\"echo \\\"TaskA\\\"\",\"resourceList\":[]}");
+        taskDefinitionLogThird.setTaskType("SHELL");
+
+        TaskDefinitionLog taskDefinitionLogFourth = taskDefinitionLogs.get(3);
+        taskDefinitionLogFourth.setCode(taskCodes.get(3));
+        taskDefinitionLogFourth.setName("TaskB");
+        taskDefinitionLogFourth.setDescription("execute TaskB");
+        taskDefinitionLogFourth.setTaskParams("{\"localParams\":[],\"rawScript\":\"echo \\\"TaskB\\\"\",\"resourceList\":[]}");
+        taskDefinitionLogFourth.setTaskType("SHELL");
+        String taskDefinitionJson = JSONUtils.toJsonString(taskDefinitionLogs);
+
+        ProxyResult ProxyResult = proxyProcessDefinitionController.createProcessDefinition(token, projectCode,
+            processDefinitionLog.getName(),
+            processDefinitionLog.getDescription(),
+            processDefinitionLog.getGlobalParams(),
+            processDefinitionLog.getLocations(),
+            processDefinitionLog.getTimeout(),
+            tenantCode,
+            taskRelationJson,
+            taskDefinitionJson,
+            PARALLEL);
+        return ProxyResult;
+    }
+    public ProxyResult shellDemo(String token, long projectCode, String tenantCode){
+
+        //get demo taskcode
+        List<Long> taskCodes = new ArrayList<>();
+        try {
+            for (int i = 0; i < 3; i++) {
+                taskCodes.add(CodeGenerateUtils.getInstance().genCode());
+            }
+        } catch (CodeGenerateUtils.CodeGenerateException e) {
+            logger.error("task code get error, ", e);
+        }
+        String taskCodeFirst = String.valueOf(taskCodes.get(0)).replaceAll("\\[|\\]", "");
+        String taskCodeSecond = String.valueOf(taskCodes.get(1)).replaceAll("\\[|\\]", "");
+        String taskCodeThird = String.valueOf(taskCodes.get(2)).replaceAll("\\[|\\]", "");
+
+        ProcessDefinitionLog processDefinitionLog = new ProcessDefinitionLog();
+        processDefinitionLog.setName("demo_shell");
+        processDefinitionLog.setDescription("Production, processing and sales of a series of processes");
+        processDefinitionLog.setGlobalParams(DemoContants.SHELL_GLOBAL_PARAMS);
+        processDefinitionLog.setLocations(DemoContants.SHELL_locations[0] + taskCodeFirst + DemoContants.SHELL_locations[1] + taskCodeSecond + DemoContants.SHELL_locations[2] + taskCodeThird + DemoContants.SHELL_locations[3]);
```

Comment:
When location is null, it will auto format in the UI.
So we don't need to set location here.